crypto/keys/secp256k1/secp256k1.go

```diff
@@ -23,42 +26,63 @@ const (
 	PubKeyName  = "tendermint/PubKeySecp256k1"
 )
 
-// PrivKey implements PrivKey.
-type PrivKey []byte
-
 // Bytes returns the byte representation of the Private Key.
-func (privKey PrivKey) Bytes() []byte {
-	return []byte(privKey)
+func (privKey *PrivKey) Bytes() []byte {
+	return privKey.Key
 }
 
 // PubKey performs the point-scalar multiplication from the privKey on the
 // generator point to get the pubkey.
-func (privKey PrivKey) PubKey() crypto.PubKey {
-	_, pubkeyObject := secp256k1.PrivKeyFromBytes(secp256k1.S256(), privKey)
-	return PubKey(pubkeyObject.SerializeCompressed())
+func (privKey *PrivKey) PubKey() crypto.PubKey {
+	_, pubkeyObject := secp256k1.PrivKeyFromBytes(secp256k1.S256(), privKey.Key)
+	pk := pubkeyObject.SerializeCompressed()
+	return &PubKey{Key: pk}
 }
 
 // Equals - you probably don't need to use this.
-// Runs in constant time based on length of the keys.
-func (privKey PrivKey) Equals(other crypto.PrivKey) bool {
-	if otherSecp, ok := other.(PrivKey); ok {
-		return subtle.ConstantTimeCompare(privKey[:], otherSecp[:]) == 1
-	}
-	return false
+// Runs in constant time based on length of the
+func (privKey *PrivKey) Equals(other crypto.PrivKey) bool {
+	return privKey.Type() == other.Type() && subtle.ConstantTimeCompare(privKey.Bytes(), other.Bytes()) == 1
 }
 
-func (privKey PrivKey) Type() string {
+func (privKey *PrivKey) Type() string {
 	return keyType
 }
 
+// MarshalAmino overrides Amino binary marshalling.
+func (privKey PrivKey) MarshalAmino() ([]byte, error) {
+	return privKey.Key, nil
+}
+
+// UnmarshalAmino overrides Amino binary marshalling.
+func (privKey *PrivKey) UnmarshalAmino(bz []byte) error {
+	*privKey = PrivKey{
+		Key: bz,
+	}
+
+	return nil
+}
+
+// MarshalAminoJSON overrides Amino JSON marshalling.
+func (privKey PrivKey) MarshalAminoJSON() ([]byte, error) {
+	// When we marshal to Amino JSON, we don't marshal the "key" field itself,
```

Comment:
Does this actually return JSON?

Reply:
Yes, AminoJSON prints bytes as base64, so it returns a JSON string here. Same behavior as current hub, e.g. https://api.cosmos.network/auth/accounts/cosmos1y3x7q772u8s25c5zve949fhanrhvmtnu484l8z